src/main/java/xyz/aadev/generitech/client/gui/upgrade/GuiUpgradeScreen.java

```diff
@@ -0,0 +1,150 @@
+        this.sides = sides;
+        this.xSize = 176;
+        this.ySize = 166;
+        this.player = player;
+        this.tileEntity = tileEntity;
+
+        machineTier = MachineTier.byMeta(tileEntity.getBlockMetadata()).getTier();
+        slot = new Rectangle(48,31, 25 ,24);
+
+    }
+
+    @Override
+    public void drawBG(int paramInt1, int paramInt2, int paramInt3, int paramInt4) {
+        bindTexture("gui/upgrade/upgrade.png");
+        drawTexturedModalRect(paramInt1, paramInt2, 0, 0, this.xSize, this.ySize);
+        if (tileEntity instanceof TileEntityMachineBase){
+            sides=((TileEntityMachineBase) tileEntity).getSides();
+        }
+    }
+
+    @Override
+    public void drawFG(int paramInt1, int paramInt2, int paramInt3, int paramInt4) {
+        // Do nothing because of not implemented
+    }
+
+
+    @Override
+    public void initGui() {
+        if (tileEntity instanceof TileEntityMachineBase){
+            this.mc.thePlayer.openContainer = this.inventorySlots;
+            this.guiLeft = (this.width - this.xSize) / 2;
+            this.guiTop = (this.height - this.ySize) / 2;
+            int Forward = ((TileEntityMachineBase) tileEntity).getForward().getIndex();
+            int Back = ((TileEntityMachineBase) tileEntity).getForward().getOpposite().getIndex();
+            int Left = 4;
+            int Right = 5;
+            if (Forward == 3) {
+                Left = 5;
+                Right = 4;
+            }
+            if (Forward == 4) {
+                Left = 3;
+                Right = 2;
+            }
+            if (Forward == 5) {
+                Left = 2;
+                Right = 3;
+            }
+
+            this.addButton(new ButtonSides(0, guiLeft + 123, guiTop + 44, sides, guiLeft + 124, guiTop + 45,tileEntity));
+            this.addButton(new ButtonSides(1, guiLeft + 123, guiTop + 20, sides, guiLeft + 124, guiTop + 21,tileEntity));
+            this.addButton(new ButtonSides(Forward, guiLeft + 123, guiTop + 32, sides, guiLeft + 124, guiTop + 33,tileEntity));
+            this.addButton(new ButtonSides(Left, guiLeft + 135, guiTop + 32, sides, guiLeft + 136, guiTop + 33,tileEntity));
+            this.addButton(new ButtonSides(Right, guiLeft + 111, guiTop + 32, sides, guiLeft + 112, guiTop + 33,tileEntity));
+            this.addButton(new ButtonSides(Back, guiLeft + 123, guiTop + 56, sides, guiLeft + 124, guiTop + 57,tileEntity));
+
+        }
+
+    }
+
+    @Override
+    protected void actionPerformed(GuiButton button) throws IOException {
+        //adds side
+        task(button.id);
+    }
+
+    private void task(int i) throws IOException {
+        Network.sendToServer(new PacketSides(sides,i,tileEntity.getPos()));
+    }
+
+    @Override
+    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
+        super.drawScreen(mouseX, mouseY, partialTicks);
+        RenderItem renderItem = mc.getRenderItem();
+        Block block = tileEntity.getBlockType();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "block" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS1481)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "block". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar.aadev.xyz/coding_rules#rule_key=squid%3AS1854)